jni-gen/src/generators/field_getter_setter.rs

```diff
@@ -0,0 +1,231 @@
+// © 2019, ETH Zurich
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// use core::num::flt2dec::Sign;
+
+use crate::{class_name::*, errors::*, utils::*};
+use jni::{
+    objects::{JClass, JObject, JValue},
+    JNIEnv,
+};
+
+fn hierarchy_field_lookup<'a>(
+    env: &JNIEnv<'a>,
+    class: &ClassName,
+    lookup_name: &str,
+) -> Result<Option<JObject<'a>>> {
+    let mut clazz = env.find_class(class.path())?;
+    while !clazz.is_null() {
+        if let Some(field) = class_field_lookup(env, clazz, lookup_name)? {
+            return Ok(Some(field));
+        }
+
+        clazz = env.get_superclass(clazz)?;
+    }
+
+    Ok(None)
+}
+
+fn class_field_lookup<'a>(
+    env: &JNIEnv<'a>,
+    clazz: JClass<'a>,
+    lookup_name: &str,
+) -> Result<Option<JObject<'a>>> {
+    let fields = env
+        .call_method(
+            clazz,
+            "getDeclaredFields",
+            "()[Ljava/lang/reflect/Field;",
+            &[],
+        )?
+        .l()?;
+
+    let num_fields = env.get_array_length(*fields)?;
+
+    for field_index in 0..num_fields {
+        let field = env.get_object_array_element(*fields, field_index)?;
+
+        let field_name = java_str_to_string(
+            &env.get_string(
+                env.call_method(field, "getName", "()Ljava/lang/String;", &[])?
+                    .l()?
+                    .into(),
+            )?,
+        )?;
+
+        if field_name == lookup_name {
+            return Ok(Some(field));
+        }
+    }
+
+    Ok(None)
+}
+
+fn generate_type_check(type_signature: &str, type_name: &str, is_result: bool) -> String {
```

Comment:
Even if it's private, can you add a doc comment to remember the `!type_signature.starts_with('L')` case and the `is_result` case?

Also, `type_name` looks confusing to me. Could you do this renaming (or abetter one) here and in the two callers?
* `type_signature` -> `expected_variable_signature`
* `type_name` -> `variable_name`

Reply:
Added the following comment and changed the variable names:
```
// Generates a runtine check that the object is of the expected class - applies only to object types (starting with L)
// If this value is also the returned value in the setter, unwrap it first (since it is of type Result<T>)
```